File: day01/python/main.py
first = []
second = []
f = open("../input.txt", "r")
for line in f:
    elements = line.split()
    first.append(int(elements[0]))
    second.append(int(elements[1]))
f.close()
first.sort()
second.sort()

result = 0
for i in range(0, len(first)):
    result = result + abs(first[i] - second[i])

print(result)

similarity = 0
for x in first:
    similarity = similarity + (x * second.count(x))
print(similarity)

File: day01/lua/main.lua
local first = {}
local second = {}
local file = io.open("../test.txt", "r")
local content = file:read("*a")
print("Print the whole file:")
io.close(file)
print(content)
file = io.open("../test.txt", "r")
local lines = file:lines()
print("Print line by line:")
local lastline
for line in lines do
	print(line)
	lastline = line
end
file:close()
print("Last line is: " .. lastline)

local beg, fin = string.find(lastline, "test", 1)
-- local uno
-- table.insert(first, uno)
-- local function mysplit(phrase, pattern)
--     local result = {}
--     if not find(phrase, pattern) do
--         return result
--     local beg, end = string.find(phrase, pattern, 1)
-- end
print("Search for the first space: ")
print(beg)
print(fin)
